### DISCUSSION TITLE
Is there any on-chain solution for creating orders?

### DISCUSSION THREAD

**Discussion by qweblessed** [ProjectOpenSea/seaport#636] [Q&A]
Hi,guys,I've just started exploring seaport protocol and I have mb a bit obvious question
According to this example:
```
            const { order, orderHash } = await createOrder(
               seller,
               zone,
               offer,
               consideration,
               0
            );

            const tx = marketplaceContract
                .connect(buyer)
                .fulfillOrder(order, toKey(0));
```
we have an order,and an order fullfiller(buyer),as far as I understand we need to store "order" struct somewhere on off-chain backend, because in other way how buyer can get this order struct to fulfill order.Is there any way how we can create orders on-chain?

**Comment by ryanio** [comment]
for on-chain you can validate the order from the seller side first and then the buyer can execute it

**Reply by mertcmutlu** [reply]
does validating the order from the seller side make it appear on OS front end?
